src/main/java/io/stargate/sgv2/jsonapi/service/schema/collections/CollectionRerankingConfig.java

```diff
@@ -35,300 +80,342 @@ public CollectionRerankingConfig(
     this(
         enabled,
         enabled
-            ? new RerankingProviderConfig(provider, modelName, authentication, parameters)
+            ? new RerankingServiceConfig(provider, modelName, authentication, parameters)
             : null);
   }
 
+  /** Returns whether reranking is enabled for this collection. */
+  public boolean enabled() {
+    return enabled;
+  }
+
+  /** Returns the reranking service configuration for this collection. */
+  public RerankingServiceConfig rerankingProviderConfig() {
+    return rerankingServiceConfig;
+  }
+
   /**
-   * Accessor for an instance to use for a default configuration for newly created collections:
-   * where no configuration defined: needs to be enabled, using "nvidia" reranking service
-   * configuration.
+   * Creates default reranking configuration for new collections.
+   *
+   * <p>When a collection is created without explicit reranking settings, this method provides a
+   * default configuration based on the reranking providers' configuration. It looks for the
+   * provider marked as default and its default model.
+   *
+   * <p>If no default provider is configured in the yaml, reranking will be disabled for new
+   * collections. Similarly, if the default provider doesn't have a default model, reranking will be
+   * disabled.
+   *
+   * @param rerankingProvidersConfig The configuration for all available reranking providers
+   * @return A default-configured CollectionRerankingConfig
    */
   public static CollectionRerankingConfig configForNewCollections(
       RerankingProvidersConfig rerankingProvidersConfig) {
-    // get the default provider from the config
-    var defaultProvider =
-        rerankingProvidersConfig.providers().entrySet().stream()
-            .filter(entry -> entry.getValue().isDefault())
-            .findFirst();
-    if (defaultProvider.isEmpty()) {
-      return new CollectionRerankingConfig(false, null);
+    // Find the provider marked as default
+    Optional<Map.Entry<String, RerankingProvidersConfig.RerankingProviderConfig>>
+        defaultProviderEntry =
+            rerankingProvidersConfig.providers().entrySet().stream()
+                .filter(entry -> entry.getValue().isDefault())
+                .findFirst();
+    // If no default provider exists, disable reranking
+    if (defaultProviderEntry.isEmpty()) {
+      return DISABLED_RERANKING_CONFIG;
     }
-    var provider = defaultProvider.get().getKey();
-    var modelName =
-        rerankingProvidersConfig.providers().get(provider).models().stream()
+
+    // Extract provider information
+    String providerName = defaultProviderEntry.get().getKey();
+    RerankingProvidersConfig.RerankingProviderConfig providerConfig =
+        defaultProviderEntry.get().getValue();
+
+    // Find the model marked as default for this provider
+    Optional<String> defaultModelName =
```

Comment:
from discussions - if there is a default provider but no model, throw IllegalState error